include/inexor/vulkan-renderer/texture.hpp

```diff
@@ -1,64 +1,114 @@
+    int mip_levels = 0;
 
-    VkImageViewCreateInfo view_create_info = {};
-
-    VkImageCreateInfo image_create_info = {};
-
-    VmaAllocation allocation = VK_NULL_HANDLE;
-
-    VmaAllocationInfo allocation_info = {};
+    VkDevice device = VK_NULL_HANDLE;
+    VkPhysicalDevice graphics_card = VK_NULL_HANDLE;
+    VkQueue data_transfer_queue = VK_NULL_HANDLE;
+    std::uint32_t data_transfer_queue_family_index = 0;
 
-    VkBufferCreateInfo create_info = {};
+    VmaAllocator vma_allocator = VK_NULL_HANDLE;
+    VmaAllocation allocation;
+    VmaAllocationInfo allocation_info;
 
-    VmaAllocationCreateInfo allocation_create_info = {};
+    VkImage image = VK_NULL_HANDLE;
+    VkImageView image_view = VK_NULL_HANDLE;
+    VkSampler sampler = VK_NULL_HANDLE;
+    VkFormat texture_image_format = VK_FORMAT_R8G8B8A8_UNORM;
 
-    VkImage image = {};
+    OnceCommandBuffer copy_command_buffer;
 
-    VkFormat format = {};
+    ///
+    void transition_image_layout(VkImage image, VkFormat format, VkImageLayout old_layout, VkImageLayout new_layout);
 
-    VkImageView image_view = {};
+    ///
+    void create_texture_image_view();
 
-    VkImageLayout image_layout = {};
+    ///
+    void create_texture_sampler();
 
-    VkDescriptorImageInfo descriptor = {};
-
-    VkSampler sampler = {};
-
-    std::uint32_t layer_count = 0;
-
-    std::uint32_t mip_levels = 0;
-
-    std::uint32_t width = 0;
-
-    std::uint32_t height = 0;
-
-    // TODO: Refactoring: Remove methods! only let VulkanTextureManager change data.
-
-    /// @brief Destroys a texture.
-    /// @param device [in] The Vulkan device.
-    /// @param vma_allocator [in] The Vulkan memory allocator handle.
-    void destroy_texture(const VkDevice &device, const VmaAllocator &vma_allocator);
-
-    /// @brief Updates the texture sampler's descriptor.
-    void update_descriptor();
+public:
+    /// Delete the copy constructor so textures are move-only objects.
+    Texture(const Texture &) = delete;
+    Texture(Texture &&other) noexcept;
+
+    /// Delete the copy assignment operator so shaders are move-only objects.
+    Texture &operator=(const Texture &) = delete;
+    Texture &operator=(Texture &&) noexcept = default;
+
+    /// @brief Creates a texture from a file.
+    /// @param device [in] The Vulkan device from which the texture will be created.
+    /// @param graphics_card [in] The graphics card.
+    /// @param vma_allocator [in] The Vulkan Memory Allocator library handle.
+    /// @param file_name [in] The file name of the texture.
+    /// @param name [in] The internal memory allocation name of the texture.
+    /// @param data_transfer_queue [in] The Vulkan data transfer queue.
+    /// @param data_transfer_queue_family_index [in] The queue family index of the data transfer queue to use.
+    Texture(const VkDevice device, const VkPhysicalDevice graphics_card, const VmaAllocator vma_allocator, const std::string &file_name, const std::string name,
```

Comment:
Consider reusing the other constructor, theres a lot of code shared between those.

Reply:
(see other comment on textures).